pre-commit/package_app_dependencies.py

```diff
@@ -276,20 +301,24 @@ def main(args):
         # Add the newly built wheels and remove the wheels no longer needed from the wheels folder
         new_wheel_paths = _copy_new_wheels(all_built_wheels, temp_dir, app_dir, app_json, pip_dependencies_key)
 
-        wheels_for_other_py_version = _parse_pip_dependency_wheels(app_json, PIP_DEPENDENCIES) \
-            if pip_dependencies_key == PIP3_DEPENDENCIES else _parse_pip_dependency_wheels(app_json, PIP3_DEPENDENCIES)
+        wheels_for_other_py_versions = list(itertools.chain.from_iterable(
+            _parse_pip_dependency_wheels(app_json, pip_dep.value)
+            for pip_dep in PipDependency if pip_dep.value != pip_dependencies_key))
 
         _remove_unreferenced_wheel_paths(app_dir=app_dir,
                                          new_wheel_paths=new_wheel_paths,
                                          existing_wheel_paths=existing_wheel_paths,
-                                         wheel_entries_for_other_py_version=wheels_for_other_py_version)
+                                         wheel_entries_for_other_py_versions=wheels_for_other_py_versions)
 
         logging.info('Updating app json with latest dependencies...')
         for pair in zip(all_built_wheels, new_wheel_paths):
             updated_app_json_wheel_entries.append(
                 AppJsonWheelEntry(pair[0].distribution, pair[1]))
-        _update_app_json(app_json, pip_dependencies_key, updated_app_json_wheel_entries, app_dir)
-    except:
+        if sorted(updated_app_json_wheel_entries) == sorted(wheels_for_other_py_versions):
+            logging.info('Same wheels found for other python versions. Not introducing a new key.')
+        else:
+            _update_app_json(app_json, pip_dependencies_key, updated_app_json_wheel_entries, app_dir)
```

Comment:
just for clarification,

this check is to prevent a new key from being added when we run the script with `pip_dependencies_key==pip39_dependencies` and we find only pure python dependencies?

Reply:
yep! without this it always generates both `pip_dependencies` and `pip39_dependencies` even if there are no differences between them since this script is called twice